File: Geeks For Geeks/Plus one/plus one.cpp
class Solution {
  public:
    vector<int> increment(vector<int> arr ,int N) {
        int i;
        for(i=N-1; i>=0; i--){
            if(arr[i]!=9){
                arr[i]+=1;
                break;
            }
            arr[i]=0;
        }
        if(i==-1){
            arr.insert(arr.begin(),1);
            return arr;
        }
        return arr;
    }
};

File: Geeks For Geeks/Plus one/Plus one.py
class Solution:
    def increment(self, arr, N):
        i=N-1
        
        while i>=0:
            if arr[i]!=9:
                arr[i]+=1
                break
            arr[i]=0
            i-=1
        if i==-1:
            return [1]+arr
        return arr
